### PR TITLE
API keys creation rake

### LINKED ISSUE #13173: Auth API on migration for existing users
Rake accepting email list that create all needed keys (see #13172).

### PR DESCRIPTION
**Pull request by amartinj**
Closes #13173

### CODE REVIEW

File: lib/tasks/user.rake [hunk 1]
```diff
@@ -61,4 +61,28 @@ namespace :user do
...
+          puts "Row #{line} is not an email: #{email}"
+          next
+        end
+
+        user = User.first(email: email)
+
+        if user.nil?
+          puts "WARN: User #{email} not found"
+        else
+          user.create_api_keys
```
Comment (javitonino) [inline lib/tasks/user.rake]: Some `rescue` with error reporting would be nice.

File: app/models/user.rb [hunk 2]
```diff
@@ -1704,6 +1704,14 @@ def destroy_account
     destroy
   end
 
+  def create_api_keys
+    return unless has_feature_flag?('auth_api')
```
Comment (javitonino) [inline app/models/user.rb]: Uhm, not sure about this. Thinking about the deployment plan, we might want to populate api keys for everyone before revealing the UI (enabling the FF).

In that sense, maybe we should allow this rake to create api keys for users without the FF?

Comment (amartinj) [reply app/models/user.rb]: yes, right

File: lib/tasks/user.rake [hunk 3]
```diff
@@ -61,4 +61,32 @@ namespace :user do
...
+
+        user = User.first(email: email)
+
+        if user.nil?
+          puts "WARN: User #{email} not found"
+        else
+          begin
+            user.create_api_keys
+          rescue => e
+            "WARN: Unable to create api keys for user with email #{email}: #{e}"
```
Comment (javitonino) [inline lib/tasks/user.rake]: Missing `puts` (and I didn't notice until acceptance 🙈)

File: lib/tasks/user.rake [hunk 4]
```diff
@@ -61,4 +61,32 @@ namespace :user do
...
+          next
+        end
+
+        user = User.first(email: email)
+
+        if user.nil?
+          puts "WARN: User #{email} not found"
+        else
+          begin
+            user.create_api_keys
```
Comment (javitonino) [inline lib/tasks/user.rake]: Please add a success message here so it's easier to track progress. Include the username and some kind of progress indicator (line variable?)

Comment (amartinj) [reply lib/tasks/user.rake]: Oook

**Comment by javitonino** [comment]
Acceptance 🍏 

I'll let the tests run again after the merge, but otherwise, 🚀

**Pull request merged**